### PR TITLE
Improve date picker fidelity

### LINKED ISSUE #3558: Date picker no longer matches the spec
From the gallery:

img_0003

From the spec (https://www.google.com/design/spec/components/pickers.html#pickers-date-pickers):

components_pickers_date1

### PR DESCRIPTION
**Pull request by abarth**
This patch updates the date picker to match the new spec.

We're still missing a fade effect when scrolling from one month to
another and we're missing the landscape layout.

Fixes #3558

### CODE REVIEW

**Comment by abarth** [comment]
@HansMuller @collinjackson

File: packages/flutter/lib/src/material/date_picker.dart [hunk 1]
```diff
@@ -430,14 +436,46 @@ class _MonthPickerState extends State<MonthPicker> {
...
   @override
   Widget build(BuildContext context) {
-    return new ScrollableLazyList(
-      key: new ValueKey<DateTime>(config.selectedDate),
-      initialScrollOffset: config.itemExtent * _monthDelta(config.firstDate, config.selectedDate),
-      itemExtent: config.itemExtent,
-      itemCount: _monthDelta(config.firstDate, config.lastDate) + 1,
-      itemBuilder: _buildItems
+    return new Stack(
+      children: <Widget> [
```
Comment (collinjackson) [inline packages/flutter/lib/src/material/date_picker.dart]: I think you have an extra space here

**Comment by collinjackson** [comment]
lgtm

**Comment by sethladd** [comment]
Thanks!!

**Pull request merged**